### PR TITLE
Added support for Azure Ad and other OpenID connect based auth providers

### PR DESCRIPTION
**Pull request by manishtiwari25**
Removed the tight coupling with auth0 and made it more common so we can use other OpenID connect based auth providers. 
I am using Azure active directory so I added these.

### CODE REVIEW

**Comment by linkdotnet** [comment]
Very good point! I will have a look tonight. Thanks @manishtiwari25

File: src/LinkDotNet.Blog.Web/Authentication/OpenIdConnect/AuthLoginManager.cs [hunk 1]
```diff
@@ -1,31 +1,35 @@
-﻿using System;
+using System;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Authentication.Cookies;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using static Raven.Client.Constants;
```
Comment (linkdotnet) [inline src/LinkDotNet.Blog.Web/Authentication/OpenIdConnect/AuthLoginManager.cs:7]: That doesn't seem right.

File: src/LinkDotNet.Blog.Web/appsettings.json [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
+	},
+	"Introduction": {
+		"Description": "A small introduction to your **blog**. Can be in markdown",
+		"BackgroundUrl": "assets/profile-background.webp",
+		"ProfilePictureUrl": "assets/profile-picture.webp"
+	},
+	"PersistenceProvider": "InMemory",
+	"ConnectionString": "",
+	"DatabaseName": "",
+	"AuthProvider": "Auth0",
```
Comment (linkdotnet) [inline src/LinkDotNet.Blog.Web/appsettings.json:25]: Should be `AuthenticationProvider`

File: src/LinkDotNet.Blog.Web/Authentication/OpenIdConnect/AuthHelper.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+using Microsoft.Extensions.Configuration;
+using System;
+
+namespace LinkDotNet.Blog.Web.Authentication.OpenIdConnect;
+
+public static class AuthHelper
+{
+    public static string GetAuthProvider(IConfiguration configuration)
```
Comment (linkdotnet) [inline src/LinkDotNet.Blog.Web/Authentication/OpenIdConnect/AuthHelper.cs:8]: I would prefer if this is a normal class that lives inside the DI - container.
Until now I used the `AppConfiguration` and `AppConfiugrationFactory` for such global settings. While this is not ideal and is open to discussion, I would prefer if we could stick to it for the time being

**Comment by linkdotnet** [comment]
I have a few points, but looks good so far!

File: src/LinkDotNet.Blog.Web/Authentication/OpenIdConnect/AuthHelper.cs (hunk 3 above)
Comment (manishtiwari25) [reply src/LinkDotNet.Blog.Web/Authentication/OpenIdConnect/AuthHelper.cs]: Moved to app configuration and AppConfiugrationFactory.

**Comment by linkdotnet** [comment]
Good job - there is one last thing: There is a [test for the `AppConfigurationFactory`](https://github.com/linkdotnet/Blog/blob/master/tests/LinkDotNet.Blog.UnitTests/Web/AppConfigurationFactoryTests.cs) that should be extended.

**Review by linkdotnet — APPROVED** [review]

**Review by linkdotnet — APPROVED** [review]

**Comment by linkdotnet** [comment]
Thanks again @manishtiwari25  - I am thinking of removing the backward compatibility fo `Auth0` if the config is not given.
I guess am fine breaking things in the code to have a cleaner code.

**Pull request merged**